### ISSUE TITLE
[JENKINS-70211] Only create stash if files exist

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/workflow-basic-steps-plugin#527]
<p>As an alternative to <a href="https://issue-redirect.jenkins.io/issue/37327" title="Allow stashing/unstashing empty sets" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/37327">JENKINS-37327</a>"><del>JENKINS-37327</del></a> it would be useful to be able to skip creating a stash altogether rather than creating an empty stash if the specified files do not exist.</p>

<p> </p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=brianjmurrell">brianjmurrell</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-70211" target="_blank">Only create stash if files exist</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Minor
<li><b>component(s)</b>: workflow-basic-steps-plugin
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 1
<li><b>watchers</b>: 2
<li><b>imported</b>: 20251215-220547
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>As an alternative to <a href="https://issues.jenkins.io/browse/JENKINS-37327" title="Allow stashing/unstashing empty sets" class="issue-link" data-issue-key="JENKINS-37327"><del>JENKINS-37327</del></a> it would be useful to be able to skip creating a stash altogether rather than creating an empty stash if the specified files do not exist.</p>

<p> </p></pre>
</details>
</details>

**Comment by killawetz** [comment]
@jglick Hi, is there any chance this will be implemented?

**Comment by jglick** [comment]
No, I think https://github.com/jenkinsci/workflow-basic-steps-plugin/pull/42 should suffice for most purposes. Pipelines with unusual needs can explicitly check whether specific files exist or not.
